src/PgAsync/Connection.php

```diff
@@ -479,6 +480,14 @@ public function query($query): Observable
     {
         return new AnonymousObservable(
             function (ObserverInterface $observer, SchedulerInterface $scheduler = null) use ($query) {
+                if ($this->connStatus === $this::CONNECTION_NEEDED) {
+                    $this->start();
+                }
+                if ($this->connStatus === $this::CONNECTION_BAD) {
+                    $observer->onError(new \Exception('Connetion failed'));
```

Comment:
spelling and should this be a specific exception?

Reply:
Spelling has been corrected.